### PR TITLE
added hint & cpp solution to usaco-1041

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/gold/usaco-1041.mdx [hunk 1]
```diff
@@ -179,4 +194,72 @@ public class Haircut {
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+// The C++ Solution uses Segement Tree instead BIT,
+// but the idea is still the same.
+
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1041.mdx:207]: can you replace the purs implementation with a class from [here](https://usaco.guide/gold/PURS?lang=cpp)?

File: solutions/gold/usaco-1041.mdx [hunk 2]
```diff
@@ -179,4 +195,72 @@ public class Haircut {
...
+	while ((1 << c) <= n + 1) { c++; }
+	c = 1 << c;
+	vector<ll> segtree(2 * c);
+	vector<ll> ans(n + 2);
+	for (int i = 1; i <= n; i++) {
+		int a;
+		fin >> a;
+		ans[a + 1] += sum(segtree, a + 1, n + 1);
+		add(segtree, a, 1);
+	}
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-1041.mdx:250]: space this out a bit?

File: solutions/gold/usaco-1041.mdx [hunk 3]
```diff
@@ -179,4 +195,72 @@ public class Haircut {
...
+
+	ll total = 0;
+	for (int i = 0; i < n; i++) {
+		total += ans[i];
+		fout << total << "\n";
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-1041.mdx:255]: endl? doesn't rly matter tho

File: solutions/gold/usaco-1041.mdx [hunk 4]
```diff
@@ -179,4 +195,72 @@ public class Haircut {
...
+	}
+
+	return 0;
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-1041.mdx:258]:
```suggestion
```

File: solutions/gold/usaco-1041.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-1041.mdx:255]: actually ryan i've changed my mind
if it's in a loop, \n is fine

File: solutions/gold/usaco-1041.mdx (hunk 1 above)
Comment (RedBlueBird) [reply solutions/gold/usaco-1041.mdx:208]: the `SumSegmentTree ` one?

Comment (SansPapyrus683) [reply solutions/gold/usaco-1041.mdx:208]: ye

**Comment by SansPapyrus683** [comment]
deal

**Comment by RedBlueBird** [comment]
oh frick totally forgot about this

**Comment by SansPapyrus683** [comment]
lmao

**Pull request merged**